### PR TITLE
Sierra: Add support for statistic group in holds and renewals.

### PR DESCRIPTION
**Pull request by EreMaijala**
Sierra 6.0 adds support for statgroup parameter that can be used to specify the "statistic group" to use in Sierra for specific functions (renewal, holds, payments).

### CODE REVIEW

**Comment by EreMaijala** [comment]
Oops, renew is not right, should be a query param.

**Comment by EreMaijala** [comment]
Should be fixed now, but I'll need to verify and can't renew the test loan again today.

File: module/VuFind/src/VuFind/ILS/Driver/SierraRest.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: I'm not sure I understand the purpose of this -- does it really make sense to add array support to the hierarchy array rather than introducing another parameter for these types of things? Would it really work/make sense to pass an array with multiple arrays, or with an array before strings? If those scenarios wouldn't work, maybe this should be approached in a different way.

**Review by demiankatz — CHANGES_REQUESTED** [review]
One question for you. In any case, I'll wait for an update on your testing before I approve anything.

File: module/VuFind/src/VuFind/ILS/Driver/SierraRest.php
Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: The whole thing of having query parameters in the URL is not a recommended practice. You're right though, this could be used to create bogus url's, but I couldn't come up with anything that wouldn't get confusing for GET requests without breaking back-compatibility.

Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: If you think it's better to add an "additional GET parameters" (that would get merged with $params for GET requests), I could go with that.

Comment (demiankatz) [reply module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: I think the additional GET parameters approach feels cleaner and would reduce the chances of invalid URL generation... but if it requires a significant amount of extra infrastructure to make it work (e.g. changing lots of method signatures and/or revising large amounts of code), I could live with this as-is since its scope of use is very limited.

File: module/VuFind/src/VuFind/ILS/Driver/SierraRest.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: Typo: becuase

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @EreMaijala, I think this is safer and more clear! Just one tiny nitpick. Happy to approve and merge this on Monday if your testing comes out successfully.

File: module/VuFind/src/VuFind/ILS/Driver/SierraRest.php
Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: Fixed!

**Comment by EreMaijala** [comment]
After testing it appears that this did not work as expected in Sierra. I'm closing this now. We can always revisit in future if necessary.

**Comment by EreMaijala** [comment]
After further inspection it seems that we will need this after all.

**Review by demiankatz — APPROVED** [review]
I can't test with Sierra, but it looks fine to me. Thanks, @EreMaijala!

**Pull request merged**
